Add patch section for USB/IP

studeo14 opened the pull request:
For some kernel versions USB/IP may require a patch in order to compile. These are fixed in current kernel versions, but some may not have the patches in their version.

Added information for these patches: stringop-trunctation in usbip_device_driver.c, and address-of-packed-member in usbip_network.c

Commit:
Add patch section for USB/IP

For some kernel versions USB/IP may require a patch in order to compile. These are fixed in current kernel versions, but some may not have the patches in their version.

Added information for these patches: stringop-trunctation in usbip_device_driver.c, and address-of-packed-member in usbip_network.c

rpasek commented on README.md:
````diff
@@ -131,6 +131,27 @@ Build USBIP tools:
 /usr/src/4.19.43-microsoft-standard/tools/usb/usbip$ sudo make install -j 12
 ```
 
````
Hi! Thanks for the commit. Q: Is this compiler message occurring because of an upgrade to GCC v8? Any ideas on why it did not occur before?

studeo14 replied:
Hi, thanks for replying! First of all, my kernel version is `4.19.104` on Ubuntu 20 with gcc `9.3.0`. The USB/IP Makefiles seem to be settings the `-Werror` flag, so all of these small warnings will make the build fail. These errors are new [features](https://gcc.gnu.org/gcc-9/changes.html#c-family) in gcc 9 so that could be why they haven't shown up before. I updated the README to reflect that as well.

dkozel replied:
I encountered the same error and these patches have resolved it (at least as far as being able to compile).

Commit:
Updated patch context

The need for the patches has been updated to reflect why they are now showing up.

dkozel commented on README.md:
````diff
@@ -131,6 +131,27 @@ Build USBIP tools:
 /usr/src/4.19.43-microsoft-standard/tools/usb/usbip$ sudo make install -j 12
 ```
 
+New [warnings](https://gcc.gnu.org/gcc-9/changes.html#c-family) were introduced in gcc 9, which can cause the USB/IP build to faild. Their causes can be patched using:
````
typo in failed

dkozel reviewed (changes requested):
This is definitely helpful and necessary and merging it as is would be an improvement. Even better would be to include the patch files in this repo and include wget and patch commands for applying the patches. While that may seem a bit excessive given we're compiling a kernel, the rest of the instructions are so straightforward and detailed that very little programming knowledge is needed to follow them.

Commit:
Added patch files to project. Added script ot apply them automatically.

Cleaned up code

Updated README to recommend using the apply_patches script

Updated README with example usages

Typo

studeo14 commented:
Just added a script to apply the patches in question. It can also handle applying a generic patch set in case new patches are needed in the future. I updated the README to reflect this.

sfxworks commented:
When using your branch:

```
quantomworks@DESKTOP-6VDCHBC:~/usbip-wsl2-instructions$ sudo bash scripts/apply_patches.sh
Found patches: ./patches/usbip-network-fix-unaligned-member-access.patch ./patches/usbip-strncpy-stringop-truncation.patch
Applying patch: ./patches/usbip-network-fix-unaligned-member-access.patch
patch -d /usr/src/4.19.104-microsoft-standard -p1 < ./patches/usbip-network-fix-unaligned-member-access.patch
patching file tools/usb/usbip/src/usbip_network.c
Hunk #1 succeeded at 62 (offset 12 lines).
patching file tools/usb/usbip/src/usbip_network.h
Applying patch: ./patches/usbip-strncpy-stringop-truncation.patch
patch -d /usr/src/4.19.104-microsoft-standard -p1 < ./patches/usbip-strncpy-stringop-truncation.patch
patching file tools/usb/usbip/libsrc/usbip_common.c
Hunk #1 FAILED at 226.
1 out of 1 hunk FAILED -- saving rejects to file tools/usb/usbip/libsrc/usbip_common.c.rej
patching file tools/usb/usbip/libsrc/usbip_device_driver.c
Hunk #1 FAILED at 103.
Hunk #2 FAILED at 122.
2 out of 2 hunks FAILED -- saving rejects to file tools/usb/usbip/libsrc/usbip_device_driver.c.rej
```
